// vim:sw=2:ts=2:sts=2:
syntax = "proto3";

package inference;

message ShapeDetectionResponse {
  repeated Point points = 1;
}

message DetectionRequest {
    bytes image = 1;
    Rectangle rect = 2;
}

service ShapeDetection {
    rpc DetectShape(DetectionRequest) returns (ShapeDetectionResponse) {}
}

service ObjectDetection {
    rpc DetectObjects(DetectionRequest) returns (stream Object) {}
}

message Point {
  int32 x = 1;
  int32 y = 2;
}

message Rectangle {
  int32 x = 1;
  int32 y = 2;
  int32 width = 3;
  int32 height = 4;
}

message Region {
  int32 x = 1;
  int32 y = 2;
  int32 width = 3;
  int32 height = 4;
}

message Object {
  string label = 1;
  Rectangle rect = 2;
}

